dialogy/plugins/text/intent_entity_mutator/__init__.py

```diff
@@ -0,0 +1,273 @@
+"""
+    IntentEntityMutatorPlugin
+"""
+
+from typing import Any, List, Optional, Dict, Iterable, Union, Callable
+
+from dialogy.base import Input, Output, Plugin, Guard
+from dialogy.types import BaseEntity, Intent
+
+from dialogy.plugins.text.intent_entity_mutator.actions_on_primitives import (
+    contain_digits,
+    is_number_absent,
+)
+
+transcript_functions_map: Dict[str, Callable[[List[str]], int]] = {
+    "is_number_absent": is_number_absent,
+    "contain_digits": contain_digits,
+}
+
+
+class IntentEntityMutatorPlugin(Plugin):
+    def __init__(
+        self,
+        dest: Optional[str] = None,
+        guards: Optional[List[Guard]] = None,
+        rules: Optional[Dict[str, List[Any]]] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(guards=guards, dest=dest, **kwargs)
+
+        if rules:
+            self.rules = rules[
+                "swap_rules"
+            ]  # Rules is a python object parsed by yaml and loaded in memory
+        else:
+            raise ValueError("Rules have not been instantiated")
+
+    def check_present_absent_other_primitives(
+        self,
+        primitive: str,
```

Comment:
Isn't this not the `primitive` but the value of the `primitive`? For e.g. it won't be `intent` but the value of `intent`?